cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -263,6 +263,23 @@ public Partition partition(TopicPartition tp) {
     return _partitionsByTopicPartition.values().stream().map(Partition::leader).collect(Collectors.toSet());
   }
 
+  /**
+   * Return a map from topic names to the number of leader replicas of each topic
+   * @param topics a set of topic names
+   * @return A map from topic names to the number of leader replicas of each topic
+   */
+  public Map<String, Long> numLeadersPerTopic(Set<String> topics) {
+    Map<String, Long> leaderCountByTopicNames = new HashMap<>(topics.size());
+    _partitionsByTopicPartition.forEach((topicPartition, partition) -> {
+      String topicName = topicPartition.topic();
+      if (topics.contains(topicName)) {
+        long leaderReplicaCount = partition.replicas().stream().filter(Replica::isLeader).count();
```

Comment:
1. Won't this always be `1`? -- i.e. doesn't a partition have 1 leader by definition?
2. How could it be possible for the number of leaders per topic to be a `Long`? Do we expect a Kafka topic to have more than `Integer.MAX_VALUE` partitions?

Can we use the following function:
```
  public Map<String, Integer> numLeadersPerTopic(Set<String> topics) {
    Map<String, Integer> leaderCountByTopicNames = new HashMap<>(topics.size());
    for (TopicPartition tp : _partitionsByTopicPartition.keySet()) {
      String topicName = tp.topic();
      if (topics.contains(topicName)) {
        leaderCountByTopicNames.merge(topicName, 1, Integer::sum);
      }
    }
    return leaderCountByTopicNames;
  }
```

Reply:
1. Yes. We can assume that a partition always has 1 leader
2. The reason why I used `Long` is that the return type of the below statement is long
`partition.replicas().stream().filter(Replica::isLeader).count();`

I will change the impl of this method to use the one you suggested